core/src/main/java/org/apache/iceberg/hadoop/HadoopFileIO.java

```diff
@@ -66,10 +66,19 @@ public OutputFile newOutputFile(String path) {
 
   @Override
   public void deleteFile(String path) {
+    deleteImpl(path, false);
+  }
+
+  @Override
+  public void deleteFolder(String path) {
+    deleteImpl(path, true);
+  }
+
+  private void deleteImpl(String path, boolean recursive) {
     Path toDelete = new Path(path);
     FileSystem fs = Util.getFs(toDelete, hadoopConf.get());
     try {
-      fs.delete(toDelete, false /* not recursive */);
+      fs.delete(toDelete, recursive);
```

Comment:
Is the directory empty here?
If not, could we first remove the files in it?

Reply:
For that we would need to list the files in the directory so that we can drop them one-by-one. I'd go for simple invoke a recursive drop here.